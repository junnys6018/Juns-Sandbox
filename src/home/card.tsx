import { Link } from 'react-router-dom';

export interface CardProps {
    title: string;
    subTitle: string;
    month: string;
    description: string;
    image: string;
    to: string;
}

export default function Card(props: CardProps) {
    return (
        <div className="home__card">
            <Link to={props.to} style={{ backgroundImage: `url(${props.image})` }}></Link>
            <div className="sm:p-6 sm:overflow-hidden">
                <div className="sm:flex sm:flex-row sm:items-center">
                    <Link
                        to={props.to}
                        className="text-pink-600 font-medium text-lg block truncate w-min max-w-full mx-auto sm:m-0"
                    >
                        "{props.title}"
                    </Link>
                    <span className="absolute sm:static right-6 sm:right-0 top-6 sm:top-0 sm:ml-auto sm:pl-4">
                        {props.month}
                    </span>
                </div>
                <h3 className="text-neutral-600 text-sm mb-3 truncate">{props.subTitle}</h3>
                <p className="text-neutral-800 line-clamp-4 sm:line-clamp-5">{props.description}</p>
            </div>
        </div>
    );
}
