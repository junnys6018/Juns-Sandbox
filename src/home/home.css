.home__hero-text {
    @apply bg-clip-text text-transparent;
    background-image: linear-gradient(90deg, rgba(255, 0, 153, 1), rgba(255, 153, 0, 1));
}

.home__link {
    @apply text-pink-500 hover:text-pink-600;
    background-image: linear-gradient(theme('colors.pink.500'), theme('colors.pink.500'));
}

.home__hero-container {
    @apply relative mb-16 xl:mb-28;
}

.home__hero-container::before {
    @apply absolute top-0 left-0 w-full -z-10 bg-no-repeat;
    height: 30rem;
    content: '';
    /* background-image: url('./background.svg'); */
}

.home__card {
    @apply text-center rounded-3xl drop-shadow-lg bg-white p-6 relative hover:scale-105 transition-transform;
}

.home__card > a {
    @apply block rounded-full bg-center bg-cover mx-auto mb-1;
    width: 6.25rem;
    aspect-ratio: 1;
}

.home__card-container {
    @apply grid mx-auto gap-y-8 mb-8;
    grid-template-columns: 90vw;
}

@screen sm {
    .home__card-container {
        @apply gap-y-10 mb-20;
        grid-template-columns: 512px;
    }

    .home__card {
        @apply grid p-0 text-left;
        grid-template-columns: 200px auto;
    }

    .home__card > a {
        @apply w-full h-full rounded-none rounded-l-3xl;
        aspect-ratio: unset;
    }
}

@screen xl {
    .home__card-container {
        @apply gap-x-16;
        grid-template-columns: 512px 512px;
    }
}
