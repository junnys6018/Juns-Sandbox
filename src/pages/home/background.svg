<svg width="1962" height="510" viewBox="0 0 1962 510" fill="none" xmlns="http://www.w3.org/2000/svg">
<circle cx="381" cy="365" r="113" fill="#FCE7F3"/>
<circle cx="837.5" cy="458.5" r="51.5" fill="#FCE7F3"/>
<circle cx="455.5" cy="116.5" r="38.5" fill="#FCE7F3"/>
<circle cx="1795.5" cy="117.5" r="166.5" fill="#FCE7F3"/>
<circle cx="1191" cy="354" r="88" fill="#FCE7F3"/>
<circle cx="1494.5" cy="393.5" r="35.5" fill="#FCE7F3"/>
<circle cx="675" cy="193" r="73" fill="#FDF2F8"/>
<circle cx="1418.5" cy="246.5" r="69.5" fill="#FDF2F8"/>
<circle cx="989.5" cy="160.5" r="85.5" fill="#FDF2F8"/>
<circle cx="114.5" cy="131.5" r="114.5" fill="#FDF2F8"/>
</svg>
