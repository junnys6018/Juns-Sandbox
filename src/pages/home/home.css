.before\:bg-image-circles::before {
    background-image: url('./background.svg');
}
