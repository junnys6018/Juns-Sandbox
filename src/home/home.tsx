import { Fragment } from 'react';
import Footer from '../footer/footer';
import Card from './card';
import ListItem from './list-item';
import './home.css';

const tempImage =
    'https://images.unsplash.com/photo-1544551763-77ef2d0cfc6c?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=2070&q=80';

export default function Home() {
    return (
        <Fragment>
            <section id="home" className="home__hero-container">
                <div className="container text-center">
                    <h1 className="font-semibold text-5xl md:text-7xl xl:text-8xl mt-16 mb-12 home__hero-text">
                        Jun's Sandbox
                    </h1>
                    <p className="mb-8 w-text mx-auto">
                        This year I challenged myself to program an interesting visualisation of a math concept or
                        algorithm each month. I have a wide interest in math and computer science so you will find demos
                        from many areas including cellular automata, path finding algorithms, fractals, ray tracing, and
                        more...
                    </p>
                    <p className="w-text mx-auto">
                        I hope you have fun tinkering with each of the demos as much as I had fun making them! To see
                        more of my work, take a look at my{' '}
                        <a
                            href="https://junnys6018.github.io/"
                            target="_blank"
                            rel="noopener noreferrer"
                            className="home__link animate-underline"
                        >
                            portfolio.
                        </a>
                    </p>
                </div>
            </section>
            <section
                id="demos"
                className="grid grid-cols-[90vw] sm:grid-cols-[512px] xl:grid-cols-[512px_512px] mx-auto gap-y-8 mb-8 sm:gap-y-10 sm:mb-20 xl:gap-x-16"
            >
                <ListItem
                    to="#"
                    title="Flock'n roll Flock'n roll Flock'n roll Flock'n roll Flock'n roll Flock'n roll"
                    subTitle="Oscillators that sync and swarm Oscillators that sync and swarm Oscillators that sync and swarm"
                    month="Jan"
                    description="Lorem ipsum dolor sit amet, consectetur adipiscing elit. Turpis aliquam et lacus, cum eleifend. Felis condimentum nam tempus suspendisse facilisis viverra tellus."
                    image={tempImage}
                />
                <ListItem
                    to="#"
                    title="Flock'n roll"
                    subTitle="Oscillators that sync and swarm"
                    month="Jan"
                    description="Lorem ipsum dolor sit amet, consectetur adipiscing elit. Turpis aliquam et lacus, cum eleifend. Felis condimentum nam tempus suspendisse facilisis viverra tellus. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Turpis aliquam et lacus, cum eleifend. Felis condimentum nam tempus suspendisse facilisis viverra tellus."
                    image={tempImage}
                />
                <ListItem
                    to="#"
                    title="Flock'n roll"
                    subTitle="Oscillators that sync and swarm"
                    month="Jan"
                    description="Lorem ipsum dolor sit amet, consectetur adipiscing elit. Turpis aliquam et lacus, cum eleifend. Felis condimentum nam tempus suspendisse facilisis viverra tellus."
                    image={tempImage}
                />
                <Card
                    to="#"
                    title="Flock'n roll Flock'n roll Flock'n roll Flock'n roll Flock'n roll Flock'n roll"
                    subTitle="Oscillators that sync and swarm Oscillators that sync and swarm Oscillators that sync and swarm"
                    month="Jan"
                    description="Lorem ipsum dolor sit amet, consectetur adipiscing elit. Turpis aliquam et lacus, cum eleifend. Felis condimentum nam tempus suspendisse facilisis viverra tellus."
                    image={tempImage}
                />
                <Card
                    to="#"
                    title="Flock'n roll"
                    subTitle="Oscillators that sync and swarm"
                    month="Jan"
                    description="Lorem ipsum dolor sit amet, consectetur adipiscing elit. Turpis aliquam et lacus, cum eleifend. Felis condimentum nam tempus suspendisse facilisis viverra tellus. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Turpis aliquam et lacus, cum eleifend. Felis condimentum nam tempus suspendisse facilisis viverra tellus."
                    image={tempImage}
                />
                <Card
                    to="#"
                    title="Flock'n roll"
                    subTitle="Oscillators that sync and swarm"
                    month="Jan"
                    description="Lorem ipsum dolor sit amet, consectetur adipiscing elit. Turpis aliquam et lacus, cum eleifend. Felis condimentum nam tempus suspendisse facilisis viverra tellus."
                    image={tempImage}
                />
            </section>
            <Footer />
        </Fragment>
    );
}
