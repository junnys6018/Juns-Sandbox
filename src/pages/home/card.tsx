import { FaArrowRight } from 'react-icons/fa';
import { Link } from 'react-router-dom';
import { CardOrListProps } from './home';

export default function Card(props: CardOrListProps) {
    return (
        <div className="group sm:grid sm:grid-cols-[200px_auto] text-center sm:text-left rounded-3xl drop-shadow-lg bg-white p-6 sm:p-0 mb-8 sm:mb-10 last:mb-0 relative transition-transform">
            <Link
                to={props.to}
                className="block rounded-full sm:rounded-none sm:rounded-l-3xl bg-center bg-cover mx-auto mb-1 w-25 h-25 sm:w-full sm:h-full"
                style={{ backgroundImage: `url(${props.image})` }}
            ></Link>
            <div className="sm:p-6 sm:overflow-hidden">
                <div className="sm:flex sm:flex-row sm:items-center">
                    <Link
                        to={props.to}
                        className="text-pink-600 font-medium text-lg block truncate w-min max-w-full mx-auto sm:m-0"
                    >
                        "{props.title}"
                    </Link>
                    <span className="absolute sm:static right-6 sm:right-0 top-6 sm:top-0 sm:ml-auto sm:pl-4">
                        {props.month}
                    </span>
                </div>
                <h3 className="text-neutral-600 text-sm mb-3 truncate sm:mr-8">{props.subTitle}</h3>
                <p className="text-neutral-800 line-clamp-4 h-24 sm:line-clamp-5 sm:h-[7.5rem] sm:mr-8">
                    {props.description}
                </p>
            </div>
            <Link
                to={props.to}
                className="inline-block opacity-0 sm:group-hover:opacity-100 absolute right-4 bottom-4 p-2 text-neutral-900 hover:text-neutral-50 hover:bg-pink-500 rounded-full transition"
            >
                <FaArrowRight size="0.875rem" />
            </Link>
        </div>
    );
}
