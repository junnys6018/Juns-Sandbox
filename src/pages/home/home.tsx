import { Fragment, useState } from 'react';
import Footer from 'components/footer/footer';
import Card from './card';
import ListItem from './list-item';
import './home.css';
import Switch from 'components/switch';
import { FaList, FaThLarge } from 'react-icons/fa';
import classNames from 'classnames';

const tempImage =
    'https://images.unsplash.com/photo-1544551763-77ef2d0cfc6c?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=2070&q=80';

export interface CardOrListProps {
    title: string;
    subTitle: string;
    month: string;
    description: string;
    image: string;
    to: string;
}

const months: CardOrListProps[] = [
    {
        to: '#',
        title: "Flock'n roll Flock'n roll Flock'n roll Flock'n roll Flock'n roll Flock'n roll",
        subTitle: 'Oscillators that sync and swarm Oscillators that sync and swarm Oscillators that sync and swarm',
        month: 'Jan',
        description:
            'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Turpis aliquam et lacus, cum eleifend. Felis condimentum nam tempus suspendisse facilisis viverra tellus.',
        image: tempImage,
    },
    {
        to: '#',
        title: "Flock'n roll",
        subTitle: 'Oscillators that sync and swarm',
        month: 'Jan',
        description:
            'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Turpis aliquam et lacus, cum eleifend. Felis condimentum nam tempus suspendisse facilisis viverra tellus. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Turpis aliquam et lacus, cum eleifend. Felis condimentum nam tempus suspendisse facilisis viverra tellus.',
        image: tempImage,
    },
    {
        to: '#',
        title: "Flock'n roll",
        subTitle: 'Oscillators that sync and swarm',
        month: 'Jan',
        description:
            'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Turpis aliquam et lacus, cum eleifend. Felis condimentum nam tempus suspendisse facilisis viverra tellus. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Turpis aliquam et lacus, cum eleifend. Felis condimentum nam tempus suspendisse facilisis viverra tellus.',
        image: tempImage,
    },
];

export default function Home() {
    const [view, setView] = useState<'GRID' | 'LIST'>('GRID');

    return (
        <Fragment>
            <section id="home" className="home__hero-container">
                <div className="container text-center">
                    <h1 className="font-semibold text-5xl md:text-7xl xl:text-8xl mt-16 mb-12 home__hero-text">
                        Jun's Sandbox
                    </h1>
                    <p className="mb-8 w-text mx-auto">
                        This year I challenged myself to program an interesting visualisation of a math concept or
                        algorithm each month. I have a wide interest in math and computer science so you will find demos
                        from many areas including cellular automata, path finding algorithms, fractals, ray tracing, and
                        more...
                    </p>
                    <p className="w-text mx-auto">
                        I hope you have fun tinkering with each of the demos as much as I had fun making them! To see
                        more of my work, take a look at my{' '}
                        <a
                            href="https://junnys6018.github.io/"
                            target="_blank"
                            rel="noopener noreferrer"
                            className="home__link animate-underline"
                        >
                            portfolio.
                        </a>
                    </p>
                </div>
            </section>
            <section
                id="demos"
                className="grid grid-cols-[90vw] sm:grid-cols-[512px] xl:grid-cols-[512px_512px] mx-auto mb-8 sm:mb-20 xl:gap-x-16"
            >
                <div className="flex flex-row xl:col-start-1 xl:col-span-2 mb-4">
                    <FaThLarge
                        className={classNames(
                            'ml-auto transition',
                            view === 'GRID' ? 'text-pink-500' : 'text-neutral-900',
                        )}
                    />
                    <Switch
                        className="mx-2.5"
                        onChange={() =>
                            setView(view => {
                                if (view === 'GRID') {
                                    return 'LIST';
                                }
                                return 'GRID';
                            })
                        }
                    />
                    <FaList
                        className={classNames('transition', view === 'LIST' ? 'text-pink-500' : 'text-neutral-900')}
                    />
                </div>
                {view === 'LIST' && months.map((props, index) => <ListItem key={index} {...props} />)}
                {view === 'GRID' && months.map((props, index) => <Card key={index} {...props} />)}
            </section>
            <Footer />
        </Fragment>
    );
}
