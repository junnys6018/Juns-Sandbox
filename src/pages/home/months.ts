import { CardOrListProps } from './home';

const tempImage =
    'https://images.unsplash.com/photo-1544551763-77ef2d0cfc6c?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=2070&q=80';


const months: CardOrListProps[] = [
    {
        to: '#',
        title: "Flock'n roll Flock'n roll Flock'n roll Flock'n roll Flock'n roll Flock'n roll",
        subTitle: 'Oscillators that sync and swarm Oscillators that sync and swarm Oscillators that sync and swarm',
        month: 'Jan',
        description:
            'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Turpis aliquam et lacus, cum eleifend. Felis condimentum nam tempus suspendisse facilisis viverra tellus.',
        image: tempImage,
    },
    {
        to: '#',
        title: "Flock'n roll",
        subTitle: 'Oscillators that sync and swarm',
        month: 'Feb',
        description:
            'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Turpis aliquam et lacus, cum eleifend. Felis condimentum nam tempus suspendisse facilisis viverra tellus. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Turpis aliquam et lacus, cum eleifend. Felis condimentum nam tempus suspendisse facilisis viverra tellus.',
        image: tempImage,
    },
    {
        to: '#',
        title: "Flock'n roll",
        subTitle: 'Oscillators that sync and swarm',
        month: 'Mar',
        description:
            'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Turpis aliquam et lacus, cum eleifend. Felis condimentum nam tempus suspendisse facilisis viverra tellus. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Turpis aliquam et lacus, cum eleifend. Felis condimentum nam tempus suspendisse facilisis viverra tellus.',
        image: tempImage,
    },
];

export default months;