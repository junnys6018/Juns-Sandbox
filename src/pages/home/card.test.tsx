import { render } from '@testing-library/react';
import Card from './card';
import { CardOrListProps } from './home';
import { MemoryRouter } from 'react-router';

test('<Card /> snapshot', () => {
    const cardProps: CardOrListProps = {
        to: '#',
        title: "Flock'n roll",
        subTitle: 'Oscillators that sync and swarm',
        month: 'Feb',
        description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Turpis aliquam et lacus, cum eleifend.',
        image: 'mock.png',
    };
    const { asFragment } = render(<Card {...cardProps} />, { wrapper: MemoryRouter });
    expect(asFragment()).toMatchSnapshot();
});
