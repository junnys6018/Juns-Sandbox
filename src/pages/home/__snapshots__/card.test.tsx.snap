// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> snapshot 1`] = `
<DocumentFragment>
  <div
    class="group sm:grid sm:grid-cols-[200px_auto] text-center sm:text-left rounded-3xl drop-shadow-lg bg-white p-6 sm:p-0 mb-8 sm:mb-10 last:mb-0 relative transition-transform"
  >
    <a
      class="block rounded-full sm:rounded-none sm:rounded-l-3xl bg-center bg-cover mx-auto mb-1 w-25 h-25 sm:w-full sm:h-full"
      href="/"
      style="background-image: url(mock.png);"
    />
    <div
      class="sm:p-6 sm:overflow-hidden"
    >
      <div
        class="sm:flex sm:flex-row sm:items-center"
      >
        <a
          class="text-pink-600 font-medium text-lg block truncate w-min max-w-full mx-auto sm:m-0"
          href="/"
        >
          "Flock'n roll"
        </a>
        <span
          class="absolute sm:static right-6 sm:right-0 top-6 sm:top-0 sm:ml-auto sm:pl-4"
        >
          Feb
        </span>
      </div>
      <h3
        class="text-neutral-600 text-sm mb-3 truncate sm:mr-8"
      >
        Oscillators that sync and swarm
      </h3>
      <p
        class="text-neutral-800 line-clamp-4 h-24 sm:line-clamp-5 sm:h-[7.5rem] sm:mr-8"
      >
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Turpis aliquam et lacus, cum eleifend.
      </p>
    </div>
    <a
      class="inline-block opacity-0 sm:group-hover:opacity-100 absolute right-4 bottom-4 p-2 text-neutral-900 hover:text-neutral-50 hover:bg-pink-500 rounded-full transition"
      href="/"
    >
      <svg
        fill="currentColor"
        height="0.875rem"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 448 512"
        width="0.875rem"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M190.5 66.9l22.2-22.2c9.4-9.4 24.6-9.4 33.9 0L441 239c9.4 9.4 9.4 24.6 0 33.9L246.6 467.3c-9.4 9.4-24.6 9.4-33.9 0l-22.2-22.2c-9.5-9.5-9.3-25 .4-34.3L311.4 296H24c-13.3 0-24-10.7-24-24v-32c0-13.3 10.7-24 24-24h287.4L190.9 101.2c-9.8-9.3-10-24.8-.4-34.3z"
        />
      </svg>
    </a>
  </div>
</DocumentFragment>
`;
